### DISCUSSION TITLE
How must a gallery be named for automatic Metadata tagging?

### DISCUSSION THREAD

**Discussion by SomeRandomDude870** [Difegue/LANraragi#482] [Q&A]
Hi, I have now tried multiple filenames and Plugins but I don´t get it.

How do I have to name my Galleries in order for them to be recognized from a Plugin?

Whenever I want to get the Metadata and go through the Plugins it will just tell me "no matching (nhentai) gallery found". The same with e(ex-hentai.

**Comment by Difegue** [comment]
👋 The Metadata plugins try various ways to match your files:

- The generated thumbnail of the files
- The `source:` tag, if it's been set by a downloader plugin beforehand
- If both of those fail, they'll use whatever's set in the "Title" metadata field.

![image](https://user-images.githubusercontent.com/8237712/121079009-e6dce900-c7d9-11eb-9d28-8c33077f2daa.png)

This "Title" field can be automatically filled in from the filename, if you set the "Filename Parsing" plugin to run automatically for new uploads:
![](https://user-images.githubusercontent.com/8237712/110715388-6cfe8900-8205-11eb-9718-d7a7cfac9bdf.png)  
(If you didn't do this, you can still catch up by running it on your files with Batch Tagging.)  


Keep in mind, if the plugins fail to find a match, you can still map them to a NH/EH URL by hand in the Edit Metadata page.  
![image](https://user-images.githubusercontent.com/8237712/121079321-463af900-c7da-11eb-8600-1456bd4b73a5.png)
